### DISCUSSION TITLE
Disable Grammarly browser extension for the source code textarea

### DISCUSSION THREAD

**Discussion by timfjord** [community/community#77830] [Repositories]
### Select Topic Area

General

### Body

[Grammarly browser extension](https://www.grammarly.com/browser) parses all `textarea`s on the page to highlight typos, if any.
But since GitHub uses the `textarea` tag to render the source code, the Grammarly extension attaches to the source code `textarea`, too, highlighting typos in the source code.
<img width="1104" alt="image" src="https://github.com/community/community/assets/471335/b0c7c48e-bfa7-4fdf-bee4-c854f12bc25c">
It doesn't really make sense for the source code, as the Grammarly browser extension is not designed to work with programming languages.
There is an option to disable the extension for the entire domain, but then you lose the ability to highlight typos when you write comments, PRs, etc.

This can be easily fixed by adding the `data-gramm="false"` property to the source code `textarea`, which would prevent the Grammarly browser extension from attaching to it.

**Comment by dakshithadissanayaka** [comment]
Sure, I can help you with the discussion quiz.



Disable Grammarly browser extension for the source code textarea.

**Answer:**

There is no option to disable Grammarly for the source code textarea specifically. However, you can disable Grammarly for the entire GitHub domain by following these steps:

1. Click the Grammarly extension icon in your browser's toolbar.
2. Select "Preferences" from the dropdown menu.
3. Click the "Sites" tab.
4. Add "github.com" to the "Sites to ignore" list.
5. Click "Save".

This will disable Grammarly for all GitHub pages, including the source code textarea.

**Comment by 0x3333** [comment]
I have Grammarly installed on my Mac as an App(Not a browser extension). The `data-gramm="false"` property doesn't work. This is very annoying. I don't want to disable it globally for Github as it is useful for writing Issues, PR, etc.

**Reply by timfjord** [reply]
I haven't tested with the app only(I have both the exception and the app installed).
But `data-gramm="false"` worked for me

**Reply by 0x3333** [reply]
With the browser extension it works, but with the native app doesn't. Thanks.
